### PR TITLE
Changes to THM volume junction

### LINKED ISSUE #25958: Make improvements to volume junction model
## Reason
Volume junctions are a critical component in systems analysis, since it is the only way to connect pipes with different orientations. Numerous users have reported deficiencies in the current formulation, such as https://github.com/idaholab/moose/discussions/25947.

## Design
This would make model improvements to the existing volume junction model (or make a new one if necessary). For a start, see https://www.sciencedirect.com/science/article/pii/S0045782519303524 for a volume junction model. This paper also includes some test problems that can be used for the junction.

## Impact
Improved systems modeling capability.

### PR DESCRIPTION
**Pull request by joshuahansel**
This makes corrections to the THM volume junction component:

- Passed full 3D velocity to numerical flux function, not just the normal component, which resulted in the incorrect thermodynamic state due to underestimation of the kinetic energy. Before, each connection was seeing a different junction pressure, for example.
- Considered contributions to tangential momentum in the volume junction due to each connection.

This required making a "3D" numerical flux function, which must return the flux in a provided normal direction.

Closes https://github.com/idaholab/moose/issues/25958.

### CODE REVIEW

**Comment by joshuahansel** [comment]
Here is an example of a comparison to some experimental data:
3pipes_open_pipe1_048
It does not match the data nearly as well as in this reference, which is where the volume junction formulation came from:

F. Daude and R. A. Berry and P. Galon.
A Finite-Volume method for compressible non-equilibrium two-phase flows in networks of elastic pipelines using the Baer–Nunziato model.
Computational Methods in Applied Mechanical Engineering. 354 (2019). 820-849.

In fact, the before results look very similar to this. However, I tried the corrections on the problem in https://github.com/idaholab/moose/discussions/25947 and saw a significant improvement.

**Comment by joshuahansel** [comment]
Because the results are only so-so, I'm not sure if I should add the above problem as a regression problem or not. But I guess how "good" the results are shouldn't be relevant for regression, only validation.

**Comment by joshuahansel** [comment]
I still need to finish this - just was never completely happy with the results, and didn't want to go through the pain of regolding if I wasn't certain if it was a complete fix.

**Comment by joshuahansel** [comment]
Note that these fixes did not solve the issues in https://github.com/idaholab/moose/discussions/28116

**Comment by joshuahansel** [comment]
I think I finally did it:
4pipes_closed_pipe1_048
This matches the results in the reference above.

**Comment by joshuahansel** [comment]
Spoiler alert: the fix was this "velocity scaling" option I added in this PR (it just needed to be fixed). The crux of this scaling is that the normal velocity in the junction cell needs to be reconstructed in an intelligent way; otherwise you get some "unwanted water-hammer-like source".

Other test problems show significant differences before and after (with "after" matching literature). These are test problems featuring shocks. I'm not sure how the differences translate into longer time-scale behavior of the junction, but I can test it.

Something that really surprised me was a case that was just one pipe connected straight to another pipe with a smaller cross-section. Our formulation without a junction gives the wrong answer. Apparently at abrupt area changes, we actually need this junction treatment:
pressure

File: modules/thermal_hydraulics/doc/content/modules/thermal_hydraulics/theory_manual/vace_model/volume_junction.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+\tilde{u}_{\text{J},i,n} = u_{i,n} - \mathcal{G}_i \left(u_{i,n} - u_{\text{J},i,n}\right) \eqc
+
+with $u_{i,n} \equiv u_{i,d} \mathbf{d}_i \cdot \mathbf{n}_{\text{J},i}$ and
+$u_{\text{J},i,n} \equiv \mathbf{u}_\text{J} \cdot \mathbf{n}_{\text{J},i}$ and
+
+!equation
+\mathcal{G}_i = \frac{1}{2} \left(1 - \text{sgn}(u_{i,n})\right)
+\min\left(\frac{\left|u_{i,n} - u_{\text{J},i,n}\right|}{c_{\text{J},i}}, 1\right) \eqc
+
+with $c_{\text{J},i} \equiv \max(c_i, c_\text{J})$.
```
Comment (licharlot) [inline modules/thermal_hydraulics/doc/content/modules/thermal_hydraulics/theory_manual/vace_model/volume_junction.md:241]: Define c.

**Comment by licharlot** [comment]
For flow area changes, the `VolumeJunction1Phase` component  with the velocity scaling now gives similar results as the `JunctionParallelChannels1Phase` (it was really off before), but is less robust. For the `conservation` test in components/junction_parallel_channels_1phase, the time step needs to be reduced by a factor of 10. 
I ran some other more complex cases and  observed also a significant reduction of the time step size when using the velocity scaling. However, this is still a significant improvement.

File: modules/thermal_hydraulics/test/tests/components/pump_1phase/tests [hunk 2]
```diff
@@ -37,7 +37,7 @@
                 Outputs/file_base=pump_pressure_check_out_orientation"
     # csvdiff made with Components/pipe1/orientation='1 0 0'
     csvdiff = 'pump_pressure_check_out_orientation.csv'
-    abs_zero = 5e-7
+    abs_zero = 1e-4
```
[the remark this answers is not in the document]
Comment (joshuahansel) [reply modules/thermal_hydraulics/test/tests/components/pump_1phase/tests:40]: This test has a bend at the pump, so my changes were expected to affect this result. This test expects a pressure rise across a pump of 95 Pa compared to 13 MPa in the pipes, so I'm ok with the 1e-4 Pa difference. The analytic prediction for the limit does not consider the bend.

File: modules/thermal_hydraulics/test/tests/problems/abrupt_area_change_liquid/tests [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    requirement = 'The system shall be able to model an abrupt area change in a liquid flow'
+    [with_junction]
+      type = CSVDiff
+      input = 'with_junction.i'
+      csvdiff = 'with_junction_vpp_FINAL.csv'
+      abs_zero = 1e-8
+      heavy = True
+      detail = 'with a junction.'
+    []
+    [without_junction]
```
[the remark this answers is not in the document]
Comment (joshuahansel) [reply modules/thermal_hydraulics/test/tests/problems/abrupt_area_change_liquid/tests:15]: I prefer to keep those results, even if they aren't good. It helps illustrate the difference between the junction models.

File: modules/thermal_hydraulics/doc/content/modules/thermal_hydraulics/theory_manual/vace_model/index.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Variable-Area Compressible Euler Flow Model
+
+The Variable-Area Compressible Euler (VACE) flow model is used to model one-dimensional,
```
[the remark this answers is not in the document]
Comment (joshuahansel) [reply modules/thermal_hydraulics/doc/content/modules/thermal_hydraulics/theory_manual/vace_model/index.md:3]: Yeah, probably "Compressible Euler" is better - I'll change it now.

File: modules/thermal_hydraulics/test/tests/problems/abrupt_area_change_liquid/tests (hunk 3 above)
[the remark this answers is not in the document]
Comment (licharlot) [reply modules/thermal_hydraulics/test/tests/problems/abrupt_area_change_liquid/tests:15]: Agree. It's just that we don't meet the requirement with that :) .

[the remark this answers is not in the document]
Comment (joshuahansel) [reply modules/thermal_hydraulics/test/tests/problems/abrupt_area_change_liquid/tests:15]: The requirement just says that we model it, not model it accurately 😄

**Review by licharlot — APPROVED** [review]

**Pull request merged**